report/pass_report.go

```diff
@@ -3,21 +3,150 @@ package report
 import (
 	_ "embed"
 	"fmt"
+	"sort"
 	"strings"
 
+	"github.com/ms-henglu/armstrong/coverage"
 	"github.com/ms-henglu/armstrong/types"
 )
 
 //go:embed pass_report.md
 var passedReportTemplate string
 
-func PassedMarkdownReport(report types.PassReport) string {
+func PassedMarkdownReport(passReport types.PassReport, coverageReport types.CoverageReport) string {
 	resourceTypes := make([]string, 0)
-	for _, resource := range report.Resources {
+	for _, resource := range passReport.Resources {
 		resourceTypes = append(resourceTypes, fmt.Sprintf("%s (%s)", resource.Type, resource.Address))
 	}
 
 	content := passedReportTemplate
 	content = strings.ReplaceAll(content, "${resource_type}", strings.Join(resourceTypes, "\n"))
+
+	var coverages []string
+	count := 0
+	for k, v := range coverageReport.Coverages {
+		count++
+
+		reportDetail := generateReport(v)
+		sort.Strings(reportDetail)
+
+		coverages = append(coverages, fmt.Sprintf(`#### <!-- %[1]v -->
+<details open>
+<summary>%[1]v (%[2]v)  %[3]v</summary>
+
+[swagger](%[4]v)
+<blockquote>
+<details open>
+<summary><span%[5]v>body(%[6]v/%[7]v)</span></summary>
+<blockquote>
+
+%[8]v
+
+</blockquote>
+</details>
+
+</blockquote>
+</details>
+
+---
+`, k.Type, k.Address, k.ApiPath, v.SourceFile, getStyle(v.IsFullyCovered), v.CoveredCount, v.TotalCount, strings.Join(reportDetail, "\n\n")))
+	}
+
+	sort.Strings(coverages)
+	content = strings.ReplaceAll(content, "${coverage}", strings.Join(coverages, "\n"))
 	return content
 }
+
+func generateReport(model *coverage.Model) []string {
+	out := make([]string, 0)
+
+	if model.Enum != nil {
+		for k, isCovered := range *model.Enum {
+			out = append(out, getEnumBoolReport(k, isCovered))
+		}
+	}
```

Comment:
If this branch is hit, the following branch should not be run ? If so, the follow if should be replaced with `else if`, or direcrly return ath the end of this branch

Reply:
thanks, I have changed the code.